common/data_format.hpp

```diff
@@ -147,21 +147,92 @@ typedef struct {
     time_point predict_target_time;   // Time that prediction targeted.
 } fast_pose_type;
 
+// Used to identify which graphics API is being used (for swapchain construction)
+enum class graphics_api { OPENGL, VULKAN, TBD };
+
+// Used to distinguish between different image handles
+enum class swapchain_usage { LEFT_SWAPCHAIN, RIGHT_SWAPCHAIN, LEFT_RENDER, RIGHT_RENDER, NA };
+
+typedef struct vk_image_handle {
+    int      file_descriptor;
```

Comment:
Why is this a file descriptor?

Reply:
The naming is a little unfortunate because it's not a Vulkan descriptor, it's Linux's native resource handle. Maybe a typedef would be better here if we ever want to better support other platforms.